src/main/java/com/commercetools/sync/categories/utils/CategoryAssetUpdateActionUtils.java

```diff
@@ -65,7 +65,9 @@
     @Nonnull
     private static List<UpdateAction<Category>> buildUpdateActionsFromOptionals(
         @Nonnull final List<Optional<UpdateAction<Category>>> optionalUpdateActions) {
-        return StreamUtils.asList(optionalUpdateActions.stream());
+        return optionalUpdateActions.stream()
```

Comment:
I would remove this method at all and all it's similar ones and have a new util that can be re-used, for example:
in `OptionalUtils`.:

````java
    public static <T> Stream<T> toStream(@Nonnull final Optional<T>... optionals) {
        return Stream.of(optionals).flatMap(OptionalUtils::toStream);
    }
````
(also refer to this related comment: https://github.com/commercetools/commercetools-sync-java/pull/275/files#diff-2571cf584818e0b8eb82fca88f068b03R39).

Ideally also, if toList is used more often we could have one `toList`:
````java
public static <T> List<T> toList(@Nonnull final Optional<T>... optionals) {
        return Stream.of(optionals).flatMap(OptionalUtils::toStream).collect(Collectors.toList());
    }
````

Reply:
i didn't get where (besides tests) we use Optional varargs to collect them to list?

Reply:
Maybe you mean smth like:
```java
public static <T> List<T> toList(@Nonnull final List<T> list) {
    return list.stream().flatMap(OptionalUtils::toStream).collect(Collectors.toList());
}
```

Reply:
> i didn't get where (besides tests) we use Optional varargs to collect them to list?

check the usage/caller of the method `buildUpdateActionsFromOptionals`:

currently it is:
````java
final List<UpdateAction<Category>> updateActions = buildUpdateActionsFromOptionals(asList(
            buildChangeAssetNameUpdateAction(oldAsset, newAsset),
            buildSetAssetDescriptionUpdateAction(oldAsset, newAsset),
            buildSetAssetTagsUpdateAction(oldAsset, newAsset),
            buildSetAssetSourcesUpdateAction(oldAsset, newAsset)
        ));
````

instead it would be:
````java
OptionalUtils.toStream(buildChangeAssetNameUpdateAction(oldAsset, newAsset),
            buildSetAssetDescriptionUpdateAction(oldAsset, newAsset),
            buildSetAssetTagsUpdateAction(oldAsset, newAsset),
            buildSetAssetSourcesUpdateAction(oldAsset, newAsset))
                     .collect(toList());
````

Reply:
and with the `toList` sugar it could even be:

````java
final List<UpdateAction<Category>> updateActions = 
OptionalUtils.toList(buildChangeAssetNameUpdateAction(oldAsset, newAsset),
            buildSetAssetDescriptionUpdateAction(oldAsset, newAsset),
            buildSetAssetTagsUpdateAction(oldAsset, newAsset),
            buildSetAssetSourcesUpdateAction(oldAsset, newAsset));
````

Reply:
but the it should be smth like:
`toMutableListFilteringEmpty`
:D

Reply:
also, about mutability and common collectors:
it thought it will be a separate PR. do you want to put it to this PR?

Reply:
https://github.com/commercetools/commercetools-sync-java/issues/276